### DISCUSSION TITLE
CDI extension with file

### DISCUSSION THREAD

**Discussion by guykatz** [microstream-one/microstream#533] [Q&A] [answered]
hi: per the docs (below) its hard to undetand what the value of the key should be:
is it a path to a file? 
full path? relative? 
including file extension? without? 
must it be in the classpath?

an example would be nice
thanks

from the docs:
`
@Inject

@ConfigProperty(name = "one.microstream.ini")

private StorageManager storageManager;
`

**Comment by rdebusscher** [comment, marked as answer]
Hi @guykatz ,

The configuration value for the  `one.microstream.ini` key is used to read configuration using the `EmbeddedStorageConfiguration.load(value)` method.

- The configuration value is a String pointing to an accessible file.
- Not class-path based
- can be relative or absolute
- must have .ini or .xml extension which determines the format (Unless a custom `ConfigurationParser` is configured)
